### PR TITLE
insights: incomplete data troubleshoot insights docs

### LINKED ISSUE #45428: insights: timeout notification points to troubleshooting section with no helpful info
The `troubleshooting` link points to `code_insights/how-tos/Troubleshooting` and there is no content on that page related to the message.  

If the docs content exists update the link to point to the correct content.  If the content does not exist create and ensure the link points to it.
 
image

/cc @joelkw @felixfbecker @vovakulikov

### PR DESCRIPTION
**Pull request by Joelkw**
cc @leonore handing the rest of this off to you!

Adding the timeout docs links. (cc @vovakulikov fyi, this should go out if the timeouts go out in the release). 

## Test plan

We should check this link works/resolves and double check the content manually. 

Closes #45428

## App preview:

- [Web](https://sg-web-troubleshoot-insights-docs.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.02 kb) | 0.00% (+0.02 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/troubleshoot-insights-docs/compare-report.html;tab=live_object) for a full comparison between the commits 4ef2b20 and 535eb04 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: doc/code_insights/references/incomplete_data_errors.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+1. Runs over extra large sets of repositories (scope your insight further to fewer repositories)
+1. Uses many boolean combinations (consider splitting into multiple data series)
+1. Runs a complex search over an especially large monorepo (consider optimizing your search query to be more specific or include more filters, like `lang:` or `path:`)
+
+In addition: 
+
+1. Timeout errors on points that have been backfilled before the creation date of the insight are more likely to occur on single, large repositories, because backfill points are calculated by running many searches, repository by repository, individually. 
+1. Timeout errors on points that have been snapshot after the creation date of the insight are more likely to occur on insights running complex searches over large numbers of repositories, because snapshot points are calculated by running a single search against the current index 
+
+## Searcher errors
```
Comment (leonore) [inline doc/code_insights/references/incomplete_data_errors.md:24]: @Joelkw were you trying to refer to the zoekt backend being down issues we've been seeing? afaik we don't surface that in the point alerts yet

**Review by chwarwick — APPROVED** [review]

**Review by vovakulikov — APPROVED** [review]

File: doc/code_insights/references/incomplete_data_errors.md (hunk 1 above)
Comment (Joelkw) [reply doc/code_insights/references/incomplete_data_errors.md]: Yeah that was it. Got it, I'll remove. Thanks.

File: doc/code_insights/references/incomplete_data_points.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+1. Runs a complex search over an especially large monorepo (consider optimizing your search query to be more specific or include more filters, like `lang:` or `file:`)
+
+In addition: 
+
+1. Timeout errors on points that have been backfilled before the creation date of the insight are more likely to occur on single, large repositories, because backfill points are calculated by running many searches, repository by repository, individually. 
+1. Timeout errors on points that have been snapshot after the creation date of the insight are more likely to occur on insights running complex searches over large numbers of repositories, because snapshot points are calculated by running a single global search against the current index.
+You can read more about this case in our [limitations](../explanations/current_limitations_of_code_insights.md#accuracy-considerations-for-an-insight-query-returning-a-large-result-set).
+
+
+If the data is available, the error alert should inform you for which times the search has timed out.
```
Comment (Joelkw) [inline doc/code_insights/references/incomplete_data_points.md:26]:
```suggestion
If the data is available, the error alert will inform you which times the search has timed out.
```

File: enterprise/internal/insights/background/queryrunner/work_handler.go [hunk 3]
```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/keegancsmith/sqlf"
 
 	"github.com/sourcegraph/log"
+
```
Comment (Joelkw) [inline enterprise/internal/insights/background/queryrunner/work_handler.go:11]:
```suggestion
```

**Pull request merged**